src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -169,10 +184,27 @@ dfsForest = Typed.dfsForest . Typed.fromAdjacencyIntMap
 -- 'forest' (dfsForestFrom [3] $ 'circuit' [1..5] + 'circuit' [5,4..1]) == 'path' [3,2,1,5,4]
 -- @
 dfsForestFrom :: [Int] -> AdjacencyIntMap -> Forest Int
-dfsForestFrom vs = Typed.dfsForestFrom vs . Typed.fromAdjacencyIntMap
+dfsForestFrom vs g = dfsForestFrom' [ v | v <- vs, hasVertex v g ] g
+
+dfsForestFrom' :: [Int] -> AdjacencyIntMap -> Forest Int
+dfsForestFrom' vs g = evalState (explore vs) mempty where
```

Comment:
Unless I'm mistaken, `mempty` here corresponds to `IntSet.empty`, so let's avoid unnecessary indirection.
```suggestion
dfsForestFrom' vs g = evalState (explore vs) IntSet.empty where
```

Reply:
There were a few other cases where I had used `mempty`, I think I caught all of them